modules/accredited_representative_portal/app/models/accredited_representative_portal/power_of_attorney_request_resolution.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module AccreditedRepresentativePortal
+  class PowerOfAttorneyRequestResolution < ApplicationRecord
+    belongs_to :power_of_attorney_request,
+               class_name: 'AccreditedRepresentativePortal::PowerOfAttorneyRequest',
+               inverse_of: :resolution
+
+    RESOLVING_TYPES = [
+      'AccreditedRepresentativePortal::PowerOfAttorneyRequestExpiration',
+      'AccreditedRepresentativePortal::PowerOfAttorneyRequestDecision'
+    ].freeze
+
+    delegated_type :resolving, types: RESOLVING_TYPES, dependent: :destroy, optional: true
```

Comment:
not optional.

Reply:
in a follow up commit we can implement all the foreign key cascades and in this case maybe a trigger. let's remove AR's handling of this with dependent